tests/data_context/abstract_data_context/test_datasource_delete.py

```diff
@@ -0,0 +1,60 @@
+import pytest
+
+from great_expectations import DataContext
+from great_expectations.data_context import CloudDataContext
+from great_expectations.data_context.types.base import DatasourceConfig
+
+
+def test_datasource_delete_removes_from_cache_and_config_data_context(
```

Comment:
Really love the tests. I foresee this becoming increasingly useful as we get closer to bridging the gap between OSS and cloud